src/org/openstreetmap/josm/data/osm/IPrimitive.java

```diff
@@ -545,4 +546,16 @@ static void resetPrimitiveChildren(IPrimitive p) {
             ((IRelation<?>) p).setMembers(null);
         }
     }
+
+    /**
+     * Get child primitives that are referred by this primitive.
+     * {@link Relation}: Members of the relation
+     * {@link Way}: Nodes used by the way
+     * {@link Node}: None
+     * @return List of child primitives
+     * @since xxx
+     */
+    default List<? extends IPrimitive> getChildren() {
+        return Collections.emptyList();
+    }
```

Comment:
I'm on the fence about having a `default` method.

An alternative would be to have a "sane" default method, e.g.
```suggestion
    default List<? extends IPrimitive> getChildren() {
        if (this instanceof IRelation) {
            return this.getMemberPrimitivesList();
        } else if (this instanceof IWay) {
            return this.getNodes();
        }
        return Collections.emptyList();
    }
```

Alternatively, we could override this method in the respective `INode`, `IWay`, and `IRelation` classes.

I don't think this is wholly necessary, but I think there might be a bunch of duplicated code where that is what is happening.